dango/dex/src/core/types.rs

```diff
@@ -0,0 +1,39 @@
+use grug::{Addr, Bounded, Udec128, Uint128, ZeroInclusiveOneExclusive};
+
+#[grug::derive(Borsh, Serde)]
+pub enum Order {
+    Market(MarketOrder),
+    Limit(LimitOrder),
+}
+
+impl Order {
+    pub fn user(&self) -> Addr {
+        match self {
+            Order::Market(order) => order.user,
+            Order::Limit(order) => order.user,
+        }
+    }
+}
+
+#[grug::derive(Borsh, Serde)]
+#[derive(Copy)]
+pub struct MarketOrder {
+    pub user: Addr,
+    /// For BUY orders, the amount of quote asset; for SELL orders, that of the
+    /// base asset.
+    pub amount: Uint128,
+    /// Max slippage percentage.
+    pub max_slippage: Bounded<Udec128, ZeroInclusiveOneExclusive>,
```

Comment:
@pacmanifold I don't feel this needs to be bounded. Say the best price is $100, I can have a max slippage of 105%, meaning the average execution price can't be worse than (1 + 1.05) * $100 = $205. It's not a sensible number, but not illegal either.